server/controllers/project.controller.js

```diff
@@ -63,12 +65,29 @@ export function getProject(req, res) {
     });
 }
 
-export function deleteProject(req, res) {
-  Project.remove({ _id: req.params.project_id }, (err) => {
-    if (err) {
-      return res.status(404).send({ message: 'Project with that id does not exist' });
+function deleteFilesFromS3(files, fileCallback) {
+  let s3ObjectUrlList = [];
+  each(files, (file, callback) => {
+    if (file.url !== undefined) {
+      s3ObjectUrlList.push(file.url);
     }
-    return res.json({ success: true });
+    callback();
+  }, (err) => {
+    fileCallback();
```

Comment:
Here this
`deleteObjectsFromS3(s3ObjectUrlList, fileCallback);`
was not used because if the project had many files on S3, only when all these files got deleted from S3 we get the `fileCallback` running. Thus, when we click on the delete button in front of a project in the "My sketches" dialogue box, the project remains there for a while and after few seconds it would disappear, when `fileCallback` gets executed upon getting response from S3 that files have been deleted.

Also once we get list of URL we need to delete from S3 we don't need anything else. We can safely delete files locally and S3 delete operation can be done after that.